src/HealthChecks.Prometheus.Metrics/PrometheusResponseWriter.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+
+namespace HealthChecks.Prometheus.Metrics
+{
+    public sealed class PrometheusResponseWriter : LivenessPrometheusMetrics
+    {
+        public static async Task WritePrometheusResultText(HttpContext context, HealthReport report)
+        {
+            var instance = new PrometheusResponseWriter();
+            instance.WriteMetricsFromHealthReport(report);
+
+            using (var resultStream = CollectionToStreamWriter(instance.Registry))
+            {
+                var content = await new StreamContent(resultStream)
+                   .ReadAsStringAsync();
+
+                context.Response.ContentType = ContentType;
+
+                await context.Response.WriteAsync(content, Encoding.UTF8);
```

Comment:
Well, works, and probably we accept this, but from the performance point use the BodyWriter is better here!

Reply:
I knew nothing about Body Writer, could you recommend me some reading?

Reply:
https://www.stevejgordon.co.uk/using-the-bodyreader-and-bodywriter-in-asp-net-core-3-0